src/ngraph_cluster_manager.cc

```diff
@@ -34,7 +34,7 @@ int NGraphClusterManager::NewCluster() {
 
 GraphDef* NGraphClusterManager::GetClusterGraph(int idx) {
   std::lock_guard<std::mutex> guard(s_cluster_graphs_mutex);
-  return s_cluster_graphs[idx];
+  return idx < s_cluster_graphs.size() ? s_cluster_graphs[idx] : nullptr;
```

Comment:
Need this change since now we might have nothing stored in `NGraphClusterManager`, so need to safely avoid accessing out-of-bounds requests